torchtitan/components/ft.py

```diff
@@ -170,9 +170,12 @@ def ft_clip_grad_norm_util(total_norm: DTensor) -> torch.Tensor:
 def maybe_semi_sync_training(
     config: JobConfig,
     ft_manager: FTManager,
-    model: torch.nn.Module,
+    model_fragments: list[torch.nn.Module],
     optimizer: torch.optim.Optimizer,
     sync_every: int,
+    should_quantize: bool,
+    fragment_sync_delay: int,
+    fragment_update_alpha: float,
```

Comment:
If these configurations are solely determined by `config` and `config` is also an argument for this API, we shouldn't have to add more arguments and should instead just use `config`.